ldk-server/src/main.rs

```diff
@@ -200,6 +205,19 @@ fn main() {
 
 							let forwarded_payment_creation_time = SystemTime::now().duration_since(UNIX_EPOCH).expect("Time must be > 1970").as_secs() as i64;
 
+							match event_publisher.publish(EventEnvelope {
+									event_type: EventType::PaymentForwarded.into(),
+									event: Some(event_envelope::Event::PaymentForwarded(events::PaymentForwarded {
+											forwarded_payment: Some(forwarded_payment.clone()),
+									})),
+							}).await{
+								Ok(_) => {},
+								Err(e) => {
+									println!("Failed to publish 'PaymentForwarded' event: {}", e);
+									return;
+								}
+							};
```

Comment:
Below, if we fail writing to storage, we wouldn't call `event_node.event_handled()`. Thus, on the next iteration we would retry the event, IIUC, causing it to be published twice.

Reply:
> Yes, it is a common pattern when both event publishing and db save has to happen in same operation. Event publishing happens before db save and duplicate events are generally expected. (even the underlying messaging systems can create duplicate events to guarantee delivery.)

Ah, is the `EventPublisher` implementation expected to de-duplicate for subscribers?

> We would ideally want some kind of circuit-breaker or delay here in case our event-queue or database are temporarily unavailable.

Do you mean a retry or failure? IIUC, seems like we currently fail if the event-queue is unavailable and retry without delay if the database isn't.